src/middlewares/validation.ts

```diff
@@ -1,29 +1,32 @@
 import { Request, Response, NextFunction } from "express";
 import User from "../models/user";
 import { verify, decode, JsonWebTokenError, sign, JwtPayload } from 'jsonwebtoken';
+import { ITokenUser } from "../controllers/check-auth.controller";
 
 export const validateToken = (req: Request, res: Response, next: NextFunction) => {
   const { token } = req.cookies;
 
+  if (!token) {
+    return res.status(401).send({ succes: false, message: 'Unauthorized: Token not found' });
+  }
+  
   try {
+    let decodedToken: ITokenUser;
 
-    const decodedToken = verify(token, process.env.JWT_KEY);
-    console.log("decodedToken:", decodedToken);
+    verify(token, process.env.JWT_KEY, (err, tokenUser: ITokenUser) => {
+      if (err) {
+        return res.clearCookie("token").status(401).send({ succes: false, message: 'Unauthorized: Bad token' });
+      }
+      decodedToken = { id: tokenUser.id };
```

Comment:
When i tested the code, tokenUser only have iat and exp variables, so decodedToken becomes { id: undefined}

tokenUser: { iat: 1679471732, exp: 1679475332 }
decodedToken: { id: undefined }